Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -1143,33 +1145,54 @@ void SoftwareGLContext::gl_read_pixels(GLint x, GLint y, GLsizei width, GLsizei
         return static_cast<GLuint>(0xffffffff * min(max(f, 0.0f), 1.0f));
     };
 
+    u8 component_size = 0;
+    switch (type) {
+    case GL_BYTE:
+    case GL_UNSIGNED_BYTE:
+        component_size = 1;
+        break;
+    case GL_SHORT:
+    case GL_UNSIGNED_SHORT:
+        component_size = 2;
+        break;
+    case GL_INT:
+    case GL_UNSIGNED_INT:
+    case GL_FLOAT:
+        component_size = 4;
+        break;
+    }
+
     if (format == GL_DEPTH_COMPONENT) {
+        // Align rows according to pack alignment
+        auto const component_alignment = max(component_size, m_pack_alignment / component_size);
+        auto const aligned_width = (width + component_alignment - 1) / component_alignment * component_alignment;
```

Comment:
I find this whole calculation confusing. I think it would be better to calculate a row-stride in bytes and then use a char pointer to skip over rows. Convert that to the required type within the switch statement.

```suggestion
        auto row_stride = (width * component_size + m_pack_alignment - 1) / m_pack_alignment * m_pack_alignment;
```

And then just before the switch statement:
```
        auto char_ptr = reinterpret_cast<char*>(pixels)[i * row_stride + j * component_size];
```

Reply:
I get what you're saying - I think I made things too complicated by introducing `component_alignment` which is not necessary at all. I've updated the PR!